app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  before_action :ensure_enabled
+
+  def index
+    scope = current_project.present? ? stage.deploy_groups : DeployGroup.all
+    render json: paginate(scope.sort_by(&:natural_order))
```

Comment:
we still fetch them from the db ... but I guess we don't have to serialize them ... I guess nobody has >1000 ... so fine either way ...